var/spack/repos/builtin/packages/ucc/0001-set_hip_path.patch

```diff
@@ -0,0 +1,44 @@
+From 36494b90cca50e23987c735b557071a0acf06531 Mon Sep 17 00:00:00 2001
+From: Afzal Patel <Afzal.Patel@amd.com>
+Date: Wed, 25 Sep 2024 15:44:11 +0000
+Subject: [PATCH] add an option to set the hip path in ROCM_PARSE_FLAGS
+ function
+
+---
+ config/m4/rocm.m4 | 6 ++++--
+ 1 file changed, 4 insertions(+), 2 deletions(-)
+
+diff --git a/config/m4/rocm.m4 b/config/m4/rocm.m4
+index c7c4ca1..db5791d 100644
+--- a/config/m4/rocm.m4
++++ b/config/m4/rocm.m4
+@@ -9,7 +9,7 @@
+ # Parse whitespace-separated ARG into appropriate LIBS, LDFLAGS, and
+ # CPPFLAGS variables.
+ AC_DEFUN([ROCM_PARSE_FLAGS],
+-[for arg in $$1 ; do
++[for arg in $1 ; do
+     AS_CASE([$arg],
+         [yes],               [],
+         [no],                [],
+@@ -20,6 +20,7 @@ AC_DEFUN([ROCM_PARSE_FLAGS],
+                                  [AC_MSG_WARN([$arg of $1 not parsed])])],
+         [*include|*include/],[AS_IF([test -d $arg], [$4="$$4 -I$arg"],
+                                  [AC_MSG_WARN([$arg of $1 not parsed])])],
++        [hip_path=*],        [$5=${arg:9}],
+         [AC_MSG_WARN([$arg of $1 not parsed])])
+ done])
+
+@@ -89,7 +90,8 @@ AS_IF([test "x$with_rocm" != "xno"],
+                           [ROCM_LIBS], [ROCM_LDFLAGS], [ROCM_CPPFLAGS], [ROCM_ROOT])],
+         [AC_MSG_NOTICE([ROCm flags given ...])
+          ROCM_PARSE_FLAGS([$with_rocm],
+-                          [ROCM_LIBS], [ROCM_LDFLAGS], [ROCM_CPPFLAGS])])
++                          [ROCM_LIBS], [ROCM_LDFLAGS], [ROCM_CPPFLAGS], [ROCM_ROOT])
++         with_rocm=$ROCM_ROOT])
+
+     SAVE_CPPFLAGS="$CPPFLAGS"
+     SAVE_LDFLAGS="$LDFLAGS"
+--
+2.34.1
+
```

Comment:
This should be submitted and reviewed upstream

Reply:
patch has been removed